roles/cluster_set_scale/tasks/main.yml

```diff
@@ -0,0 +1,80 @@
+# See meta/main.yml for a list of dependency roles that will be executed prior to this role
+---
+- name: Ensure parameters have been set
+  fail: msg="Bailing out. This role requires 'machineset_instance_type' and 'scale' to be defined"
+  when: not machineset_instance_type or not scale
+
+- name: Get current replicas of machinesets with type {{ machineset_instance_type }}
+  command:
+    oc get machineset -n openshift-machine-api
+    -o=jsonpath='{range .items[?(@.spec.template.spec.providerSpec.value.instanceType=="{{ machineset_instance_type }}")]}{.spec.replicas}{"\n"}{end}'
+  register: machineset_replicas
+
+- name: Calculate sum of replicas of machinesets with type {{ machineset_instance_type }}
+  set_fact:
+    current_replicas_sum: "{{ machineset_replicas.stdout_lines | map('int') | sum(start=0) }}"
+
+- name: Log sum result
+  debug:
+    msg: "Current sum of replicas is {{ current_replicas_sum }}, wanted {{ scale }}"
+
+- when: current_replicas_sum != scale
+  name: Change all {{ machineset_instance_type }} machinesets replicas to have sum {{ scale }}
+  block:
+  - name: Get the names of existing worker machinesets with type {{ machineset_instance_type }}
+    command:
+      oc get machineset -n openshift-machine-api -o=jsonpath='{range .items[?(@.spec.template.spec.providerSpec.value.instanceType=="{{ machineset_instance_type }}")]}{.metadata.name}{"\n"}{end}'
+    register: oc_get_machinesets
+    failed_when: not oc_get_machinesets.stdout
+
+  - name: Set the replicas of all {{ machineset_instance_type }} machinesets to 0 except the first
+    command: |
+      oc patch machineset -n openshift-machine-api {{ item }} --patch '{"spec": {"replicas": 0 }}' --type merge
+    loop: "{{ oc_get_machinesets.stdout_lines[1:] }}"
+
+  - name: Get first machineset
+    set_fact:
+      first_machineset: "{{ oc_get_machinesets.stdout_lines[0] }}"
+
+  - name: Set the replicas of the first {{ machineset_instance_type }} machineset ({{ first_machineset }}) to {{ scale }}
+    command: >
+      oc patch machineset -n openshift-machine-api {{ first_machineset }}
+      --patch '{"spec": {"replicas": {{ scale }} }}' --type merge
+
+- block:
+  - name: Wait for all machinesets with type {{ machineset_instance_type }} to be ready
```

Comment:
nice, this wait for the Node to be running, I wasn't sure!
Could you add a comment saying that the `.status.availableReplicas` counter tracks the # of nodes running, and not only the # of `Machines`? (I though it would be the latest only)